### PR TITLE
[11.x] Add pan format for String helper

### PR DESCRIPTION
**Pull request by alexanderkroneis**
This PR introduces another helper function on `Str` and `Stringable`. It formats a given string into a PAN.

PAN is a Primary Account Number which is used for credit card numbers and IBANs.

```php
Str::pan('AT866000031866916134') // AT86 6000 0318 6691 6134
```

```php
str('AT866000031866916134')->pan() // AT86 6000 0318 6691 6134
```

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -472,6 +472,19 @@ public static function unwrap($value, $before, $after = null)
...
+     * Formats a given string into a valid PAN (Primary Account Number, which is used for  e.g. IBAN and credit card numbers).
+     *
+     * @param  string  $value
+     * @return string
+     */
+    public static function pan($value)
+    {
+        $value = (string) $value;
+
+        return join(' ', str_split($value, 4));
```
Comment (crynobone) [inline src/Illuminate/Support/Str.php:485]:
```suggestion
        return implode(' ', str_split($value, 4));
```

Laravel may have `Collection::join()` and `Arr::join()`. But when using PHP built-in function we always opt for `implode()` as `join()` is an alias: https://www.php.net/manual/en/function.join.php

**Comment by Jubeki** [comment]
I would love to the following values also being correctly transformed into sets of 4 symbols, so that even with spaces, the correct result is given:

```
AT8 660 000 318 669 161 34
AT86 6000 0318 6691 6134
```

I normally used the following code for that:

```
trim(chunk_split(
    str_replace(' ', '', $value),
    4,
    ' '
))
```

File: src/Illuminate/Support/Str.php (hunk 1 above)
Comment (alexanderkroneis) [reply src/Illuminate/Support/Str.php]: Thank you. 😄

**Comment by alexanderkroneis** [comment]
> I would love to the following values also being correctly transformed into sets of 4 symbols, so that even with spaces, the correct result is given:
> 
> ```
> AT8 660 000 318 669 161 34
> AT86 6000 0318 6691 6134
> ```
> 
> I normally used the following code for that:
> 
> ```
> trim(chunk_split(
>     str_replace(' ', '', $value),
>     4,
>     ' '
> ))
> ```

Thanks, that a great input!

**Comment by taylorotwell** [comment]
I think you could macro this into your project if needed. 👍

**Pull request closed without merging**